### PR TITLE
Introduce Dimensions::Item#warehouse_item_id

### PR DESCRIPTION
**Pull request by peteglondon**
We need a way to identify the same unique item of content.
Until now we have been using base_path to identify an
individual piece of content.

If the base path changes, we will lose the link between the
editions. All metrics collected from GA will be linked to the
individual editions.

The warehouse_item_id will be populated by the Etl process and
will link different versions together.

We can then use this column to join the latest item to
all previous editions, and summarise the metrics across all
editions.

This PR populates the warehouse_item_id during the Etl process and uses the warehouse_item_id to join Dimension::Items to the latest version of Dimensions::Item for the index query.

It also adds a migration to populate the column, and once a it's populated, add the not null constraint.

Guides and Travel Advice (multipart types) 

This implements [ADR-14](https://github.com/alphagov/content-performance-manager/pull/924) and [ADR-15](https://github.com/alphagov/content-performance-manager/pull/939)

### CODE REVIEW

**Comment by peteglondon** [comment]
[(5) Index page: show 100 first content items (EPIC 5)](https://trello.com/c/HvQXxa4z/659-5-index-page-show-100-first-content-items-epic-5)

**Comment by tijmenb** [comment]
Perhaps a different name for `content_uuid` might be better because this isn't a UUID according to the spec, and because `content_id` is also a UUID.

Perhaps something like `compound_id`, `warehouse_item_id`, `grain_id`, `internal_page_id`?

File: spec/models/dimensions/item_spec.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@
   it { is_expected.to validate_presence_of(:base_path) }
   it { is_expected.to validate_presence_of(:publishing_api_payload_version) }
   it { is_expected.to validate_presence_of(:schema_name) }
+  it { is_expected.to validate_presence_of(:publishing_api_payload_version) }
```
[the remark this answers is not in the document]
Comment (peteglondon) [reply spec/models/dimensions/item_spec.rb]: Should have been warehouse_item_id, corrected now.

File: db/migrate/20180921122451_add_index_dimensions_items_content_uuid.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class AddIndexDimensionsItemsContentUuid < ActiveRecord::Migration[5.2]
+  def change
+    add_index :dimensions_items, :content_uuid, name: 'index_dimensions_items_content_uuid'
+    add_index :dimensions_items, %i[content_uuid latest], name: 'index_dimensions_items_content_uuid_latest'
```
[the remark this answers is not in the document]
Comment (peteglondon) [reply db/migrate/20180921122451_add_index_dimensions_items_content_uuid.rb]: We can add this index in a later migration once the column is populated.

File: app/models/dimensions/date.rb [hunk 3]
```diff
@@ -3,7 +3,8 @@ class Dimensions::Date < ApplicationRecord
 
   scope :between, ->(from, to) { where('date BETWEEN ? AND ?', from, to) }
 
-  def self.build(date)
+  def self.build(date_or_str)
```
[the remark this answers is not in the document]
Comment (peteglondon) [reply app/models/dimensions/date.rb]: Refactored to let the factory sort it out.

**Pull request merged**